src/main/java/org/jenkins/plugins/lockableresources/actions/LockableResourcesRootAction.java

```diff
@@ -126,6 +126,26 @@ public void doReserve(StaplerRequest req, StaplerResponse rsp)
 		rsp.forwardToPreviousPage(req);
 	}
 
+	public void doSteal(StaplerRequest req, StaplerResponse rsp)
+		throws IOException, ServletException {
+		Jenkins.getInstance().checkPermission(RESERVE);
+
+		String name = req.getParameter("resource");
+		LockableResource r = LockableResourcesManager.get().fromName(name);
+		if (r == null) {
+			rsp.sendError(404, "Resource not found " + name);
+			return;
+		}
+
+		List<LockableResource> resources = new ArrayList<>();
+		resources.add(r);
+		String userName = getUserName();
+		if (userName != null)
+			LockableResourcesManager.get().steal(resources, userName);
```

Comment:
Comparing to code around https://github.com/jenkinsci/lockable-resources-plugin/pull/144/files#diff-267821931388d54e1318e5a22c6a6557R146 there is less access-permissions and related sanity checking here.

Reply:
FWIW, at https://github.com/jimklimov/lockable-resources-plugin/tree/privateBuild I track a merger of these two PRs. They seem non-conflicting in things they change logically, but there are new different routines added in the same places so automatic git merger fails to put the two together.

Reply:
That was functionally copied from the reserve function right above it with one function call alteration (.reserve -> .steal). That said I'm not sure whether it's more appropriate to do .checkPermission(RESERVE) or .checkPermission(UNLOCK) for that function. Any input would be welcome